app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,84 @@
+class TripsController < ApplicationController
+  def index
+    if params[:passenger_id]
+      @passenger = Passenger.find_by(id: params[:passenger_id])
+      @trips = @passenger.trips
+    else
+      @trips = Trip.all
+    end
+  end
+
+  def show
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      redirect_to trips_path
+      return
+    end
+  end
+
+  def new
+    passenger = nil
+    if !params[:passenger_id].nil?
+      passenger = Passenger.find_by(id: params[:passenger_id])
+    end
+    @trip = Trip.create_new(passenger)
+  end
+
+  def create
+    @trip = Trip.new(trip_params.merge(:rating => nil)) 
```

Comment:
You're not using a form to create the new trip, so you shouldn't use `trip_params`.  Instead you should get the passenger from the route, and automatically assign a driver, cost etc.

⚠️ This is broken BTW